src/common/buffer.cc

```diff
@@ -1350,7 +1350,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
 
   // -- buffer::list --
 
-  buffer::list::list(list&& other)
+  buffer::list::list(list&& other) noexcept
     : _buffers(std::move(other._buffers)),
```

Comment:
> agree with you that it's safe to make sure that every line of the function should be noexcept if the function is noexcept. but i think the noexcept also defines an "exception boundary".
> 
> what if we can/do not handle the bad_alloc exception? can we add the noexcept specifier to the function which throws bad_alloc for better performance?

it had been my understanding that noexcept actually turned off the bookkeeping that supports exception handling and that any exception thrown would lead to std::terminate, but it sounds like your interpretation is closer to the [spec](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2012/n3337.pdf):

> [15.4.9] Whenever an exception is thrown and the search for a handler (15.3) encounters the outermost block of a function with an _exception-specification_ that does not allow the exception, then [...] the function `std::terminate()` is called (15.5.1).

[cppreference](http://en.cppreference.com/w/cpp/language/noexcept_spec) rephrases it:

> If a search for a matching exception handler leaves a function marked `noexcept` or `noexcept(true)`, `std::terminate` is called immediately.